demo/body3d_two_stage_video_demo.py

```diff
@@ -0,0 +1,357 @@
+
+    # First stage: 2D pose detection
+    print('Stage 1: 2D pose detection.')
+
+    person_det_model = init_detector(
+        args.det_config, args.det_checkpoint, device=args.device.lower())
+
+    pose_det_model = init_pose_model(
+        args.pose_detector_config,
+        args.pose_detector_checkpoint,
+        device=args.device.lower())
+
+    assert pose_det_model.cfg.model.type == 'TopDown', 'Only "TopDown"' \
+        'model is supported for the 1st stage (2D pose detection)'
+
+    pose_det_dataset = pose_det_model.cfg.data['test']['type']
+
+    pose_det_results_list = []
+    next_id = 0
+    pose_det_results = []
+    for frame in video:
+        pose_det_results_last = pose_det_results
+
+        # test a single image, the resulting box is (x1, y1, x2, y2)
+        mmdet_results = inference_detector(person_det_model, frame)
+
+        # keep the person class bounding boxes.
+        person_det_results = process_mmdet_results(mmdet_results,
+                                                   args.det_cat_id)
+
+        # make person results for single image
+        pose_det_results, _ = inference_top_down_pose_model(
+            pose_det_model,
+            frame,
+            person_det_results,
+            bbox_thr=args.bbox_thr,
+            format='xyxy',
+            dataset=pose_det_dataset,
+            return_heatmap=False,
+            outputs=None)
+
+        # get track id for each person instance
+        pose_det_results, next_id = get_track_id(
+            pose_det_results,
+            pose_det_results_last,
+            next_id,
+            use_oks=args.use_oks_tracking,
+            tracking_thr=args.tracking_thr,
+            use_one_euro=args.euro,
+            fps=video.fps)
+
+        pose_det_results_list.append(copy.deepcopy(pose_det_results))
+
+    # Second stage: Pose lifting
+    print('Stage 2: 2D-to-3D pose lifting.')
+
+    pose_lift_model = init_pose_model(
+        args.pose_lifter_config,
+        args.pose_lifter_checkpoint,
+        device=args.device.lower())
+
+    assert pose_lift_model.cfg.model.type == 'PoseLifter', \
+        'Only "PoseLifter" model is supported for the 2nd stage ' \
+        '(2D-to-3D lifting)'
+    pose_lift_dataset = pose_lift_model.cfg.data['test']['type']
+
+    if args.out_video_root == '':
+        save_out_video = False
+    else:
+        os.makedirs(args.out_video_root, exist_ok=True)
+        save_out_video = True
+
+    if save_out_video:
+        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
+        fps = video.fps
+        writer = None
+
+    # transform the predicted 2D keypoints to the keypoints convention of pose
+    # lift dataset
+    for pose_det_results in pose_det_results_list:
+        for res in pose_det_results:
+            keypoints = res['keypoints']
+            res['keypoints'] = dataset_transform(keypoints, pose_det_dataset,
+                                                 pose_lift_dataset)
+
+    # prepare for temporal padding
```

Comment:
Is it possible to wrap temporal padding and index inferring (L261-L276 and L281-288) into helper functions and put them into inference_3d? I think it's too detailed for a demo script.